shiftcommerce/flex-ruby-gem
# Issue/172 update test order mode

### The issue #172 

Once the deployment engine is updated to use the app setup api then apps will be built on heroku using the app.json schema. This means that all values must be a string or object and cannot be a boolean. The environment configurations that are submitted in the admin panel will be a string e.g TEST_ODER_MODE="true". In the ruby gem we are currently checking `test_order_mode == true`, once this becomes `test_order_mode == "true"` it will become false and we will not be setting test orders.

### The fix

Update the conditional to checkout `== (true || "true")`. This way the gem will work before the deployment engine is released and after.

### QA

See QA steps in this PR - https://github.com/shiftcommerce/matalan-rails-site/pull/3339
